### PR TITLE
Fix bad transaction support

### PR DESCRIPTION
**Pull request by prolic**
- make use of $isolated operator
- collection name is extracted from stream name
- update readme

fixes: https://github.com/prooph/event-store-mongodb-adapter/issues/26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,12 +21,13 @@ The default write concern for this adapater is acknowledged and journaled (['w'
...
 This adapter does not use the MongoDB ObjectId for its primary key, instead a UUID (string) is used.
 
-When using MongoDB's sharding feature, simply set the "_id" as shard key.
+We recommend the AggregateStreamStrategy as the best strategy to use with this adapter.
 
-For more information about sharding in MongoDB check the MongoDB manual.
- 
+This adapter uses die `$isolated` operator to achieve transaction safety for a single collection.
+Keep in mind, that `$isolated` does not work with sharded clusters. Therefore it's not safe to use this adapter
+in a shared cluster environment, as we MongoDB can't guarantee transaction safety.
```
Comment (bweston92) [inline README.md]: "As we MongoDB" should be "As with MongoDB"?

File: src/MongoDbEventStoreAdapter.php [hunk 2]
```diff
@@ -187,12 +169,16 @@ public function create(Stream $stream)
      */
     public function appendTo(StreamName $streamName, array $streamEvents)
     {
+        $this->currentStreamName = $streamName;
```
Comment (codeliner) [inline src/MongoDbEventStoreAdapter.php:176]: We need to make sure that only one stream can be modified per transaction. So we should check here that either `$this->currentStreamName` is `null` or the same as `$streamName`. On transaction **commit/rollback** `$this->currentStreamName` should be reset to `null`

File: src/MongoDbEventStoreAdapter.php [hunk 3]
```diff
@@ -322,7 +311,7 @@ public function commit()
             ]
         );
 
-        $this->getUpdateBatch()->execute();
+        $updateBatch->execute();
 
         $this->transactionId = null;
     }
```
Comment (codeliner) [inline src/MongoDbEventStoreAdapter.php:322]: Also set `$this->currentStreamName` to `null`. See comment above.

File: src/MongoDbEventStoreAdapter.php [hunk 4]
```diff
@@ -334,24 +323,27 @@ public function commit()
...
                 'transaction_id' => $this->transactionId
             ],
             'limit' => 0
         ]);
 
-        $this->getDeleteBatch()->execute();
+        $deleteBatch->execute();
 
         $this->transactionId = null;
     }
```
Comment (codeliner) [inline src/MongoDbEventStoreAdapter.php:344]: Also set `$this->currentStreamName` to `null`. See comment above.

**Comment by codeliner** [comment]
@prolic Great job :+1: . I think the BC break is ok here as we really need to change the behavior of the adapter. A new major version will underline the importance of the adjustments.
PR looks good. Just a minor note. Should be easy to fix.

**Comment by prolic** [comment]
I totally missed the checks for current stream name, just slipped through my mind ;)

**Comment by codeliner** [comment]
Seems like travis won't start this build. However, I'm going to merge the PR and trigger a new build with the merge.

**Pull request merged**